File: storage-sdk/src/main/java/com/tencent/wink/storage/winkdb/converter/StringListTypeConverter.java
package com.tencent.wink.storage.winkdb.converter;

import androidx.annotation.Nullable;
import androidx.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.List;

/**
 * Created by walkerzpli on 2021/9/22.
 */
public class StringListTypeConverter implements IListTypeConverter<String> {

    @TypeConverter
    @Nullable
    @Override
    public String converter(@Nullable List<String> list) {
        return new Gson().toJson(list);
    }

    @TypeConverter
    @Nullable
    @Override
    public List<String> revert(@Nullable String str) {
        Type type = new TypeToken<List<String>>() {
        }.getType();
        return new Gson().fromJson(str, type);
    }

}

File: storage-ktx/src/main/java/com/demo/storage_ktx/converter/StringListTypeConverter.kt
package com.demo.storage_ktx.converter

import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken

/**
 * Created by walkerzpli on 2021/9/22.
 */
class StringListTypeConverter : IListTypeConverter<String> {

    @TypeConverter
    override fun converter(list: List<String>?): String {
        return Gson().toJson(list)
    }

    @TypeConverter
    override fun revert(str: String?): List<String> {
        val type = object : TypeToken<List<String>>() {}.type
        return Gson().fromJson(str, type)
    }
}